sources/packages/backend/apps/queue-consumers/src/processors/schedulers/cas-integration/cas-supplier-integration.scheduler.ts

```diff
@@ -0,0 +1,80 @@
+import { InjectQueue, Process, Processor } from "@nestjs/bull";
+import { QueueService } from "@sims/services/queue";
+import { QueueNames } from "@sims/utilities";
+import {
+  InjectLogger,
+  LoggerService,
+  ProcessSummary,
+} from "@sims/utilities/logger";
+import { Job, Queue } from "bull";
+import { BaseScheduler } from "../base-scheduler";
+import { CASSupplierIntegrationService } from "../../../services";
+import {
+  logProcessSummaryToJobLogger,
+  getSuccessMessageWithAttentionCheck,
+} from "../../../utilities";
+
+@Processor(QueueNames.CASSupplierIntegration)
+export class CASSupplierIntegrationScheduler extends BaseScheduler<void> {
+  constructor(
+    @InjectQueue(QueueNames.CASSupplierIntegration)
+    schedulerQueue: Queue<void>,
+    queueService: QueueService,
+    private readonly casSupplierIntegrationService: CASSupplierIntegrationService,
+  ) {
+    super(schedulerQueue, queueService);
+  }
+
+  /**
+   * Scheduler for CAS supplier information process.
+   * Checks for pending supplier information for students, request it to CAS and update the data to CAS supplier table.
+   * @returns process summary.
+   */
+  @Process()
+  async processCASSupplierInformation(job: Job<void>): Promise<string[]> {
+    const processSummary = new ProcessSummary();
+
+    try {
+      this.logger.log(
+        `Processing CAS supplier integration job ${job.id} of type ${job.name}.`,
+      );
+      processSummary.info("Executing CAS supplier integration...");
+      const pendingCASSuppliers =
+        await this.casSupplierIntegrationService.getStudentsToUpdateSupplierInformation();
+      const serviceProcessSummary = new ProcessSummary();
```

Comment:
Please move `serviceProcessSummary` inside the `if (pendingCASSuppliers.length)` where it will be used.